### ISSUE TITLE
Have non-modal dialog close when owner closes?

### ISSUE THREAD

**Issue by metal450** [FantasticFiasco/mvvm-dialogs#107]
I'm using this to show a non-modal dialog, i.e.

    App.DialogSvc.Show<Something>(this, somethingVm);

The application is a tabbed document editor, where the non-modal dialog shown above is a pop-up with some tools that apply to the current document.  The owner is the VM representing the document.  The issue: when the current document tab (of the owner VM) is closed, the non-modal dialog remains open.  Under old WinForms, when I'd open a non-modal window & pass the current window as its owner, closing the current window would implicitly close the child.  It's fairly obvious why this doesn't happen with MVVM, but I'm at a loss as to how a similar behavior could be achieved with MVVM.

Basically, I'm trying to figure out a way - either manually or automatically - to have the non-modal dialog close when the view of the VM from which it was opened closes.

**Comment by FantasticFiasco** [comment]
Hi!

I'll see if we can't get a sample showing this scenario. But since it's a National day here in Sweden, it will probably be a few days until I have a working sample.

**Comment by metal450** [comment]
No prob - thanks for the quick reply!

**Comment by FantasticFiasco** [comment]
I've created a very small sample [here](https://github.com/FantasticFiasco/mvvm-dialogs-sample-close-non-modal-dialog). Closing a non-modal dialog using MVVM is not something that is supported by this library, but it can be achieved by attached properties you write on your own, as demonstrated in the sample.

Would you consider this to be a sufficient solution for your problem?

**Comment by metal450** [comment]
Perfect solution!  Thank you again!

**Comment by FantasticFiasco** [comment]
I wasn't happy with the proposed solution. The fact that you had to implement your own helper class for closing the non-modal dialog just bugged me.

I've now released a new version of this library (v7.1.0), a version where you can close your dialog. I've updated the sample linked above, please have a look at it and use the new version of MVVM Dialogs in your code.

I've also mentioned you in `CHANGELOG.md` since you are the reason this feature got implemented.

Anyway, best of luck on all your future endeavours, and don't hesitate to open a new issue if you find anything to your disliking.

**Comment by metal450** [comment]
Very cool, thanks again!
